File: Programa/logica/ficha.py
import enum

#Representan los estados de la ficha
class EstadoFicha(enum.Enum):
	casa = 0
	tablero = 1
	pasillo = 2
	gano = 3

class Ficha:
	estado = ""
	movimientos = 0
	posicion = 0
	color = ""

	def __init__(self,pColor = None):
		self.color = pColor
		self.movimientos = 0
		self.posicion = 0
		self.estado = EstadoFicha.casa

	#Mueve la ficha a una posicion
	def sumarMovimiento(self):
		self.movimientos += 1

	def cambiarPosicion(self, nuevaPosicion):
		self.posicion = nuevaPosicion

	#Regresa a casa la ficha
	def regresarCasa(self):
		posicion = 0
		self.movimientos = 0
		self.estado = EstadoFicha.casa

	def toString(self):
		return "Estado: " + self.estado.name + " Posicion: " + str(self.movimientos)

File: Programa/logica/Ficha.js
const EstadoFicha = require('./EstadoFicha')

// const EstadoFicha = {
//     CASA: 0,
//     TABLERO : 1,
//     PASILLO : 2,
//     GANO : 3
// }


module.exports = class Ficha {
    constructor(color = null){
        
        this.color = color; 
        this.movimientos = 0;
        this.posicion = 0;
        this.estado = EstadoFicha.CASA;
    }

    sumarMovimientos(){
        this.movimientos +=1;
    }

    cambiarPosicion(nuevaPos){
        this.posicion = nuevaPos;
    }

    regresarCasa(){
        this.posicion = 0;
        this.movimientos =0;
        this.estado = EstadoFicha.CASA;
    }

    toString(){
        console.log("Estado: ",this.estado,"Posicion: "+ this.movimientos);
    }

}

